game-state-service

```diff
@@ -177,6 +190,8 @@ class GameStateService(Gio.Application):
             connection.emit_signal(None, path, iface, 'reset', None)
             invocation.return_value(None)
 
+        self.release()
```

Comment:
On further reflection, it might be good to structure the method like this:
```py
self.hold()
try:
    # ... do all stuff here
finally:
    self.release()
```
just to be robust against future early returns, or unexpected exceptions.

Reply:
Makes sense. I've updated the second patch.